Sources/AWSLambdaRuntimeCore/LambdaContext.swift

```diff
@@ -49,40 +50,118 @@ extension Lambda {
 extension Lambda {
     /// Lambda runtime context.
     /// The Lambda runtime generates and passes the `Context` to the Lambda handler as an argument.
-    public final class Context: CustomDebugStringConvertible {
+    public struct Context: BaggageContext.Context, CustomDebugStringConvertible {
+        private var storage: _Storage
+
+        final class _Storage {
+            var baggage: Baggage
+
+            let invokedFunctionARN: String
+            let deadline: DispatchWallTime
+            let cognitoIdentity: String?
+            let clientContext: String?
+
+            var _logger: Logger
```

Comment:
Why has logger the underscore in its front?

Reply:
Could this also be a `let`? I can't see anything that modifies it...

Reply:
Oh good point, yeah it should be modifiable if we go with the get+set Context style (which I do think is superior) along with the CoW contexts. 

Will fix thx!

Reply:
> Why has logger the underscore in its front?

Because the actual logger always has to be created on demand via:

```
var logger: Logger {
             self.storage._logger.with(self.baggage)
```

(that's the contract of `Context`), because then it always has the latest baggage.

Reply:
Okay, if you want to make this point explicit, I'd be in favor of a code comment on `Storage._logger`, explaining your thinking.

Reply:
Okey will do.